oak_idl/src/lib.rs

```diff
@@ -20,120 +20,39 @@ extern crate alloc;
 
 use alloc::vec::Vec;
 use core::fmt::Debug;
-use serde::{Deserialize, Serialize};
 
 pub mod utils;
 
 #[derive(Debug)]
-pub enum ClientError {
+pub enum Error {
+    /// The request message could not be deserialized correctly.
```

Comment:
If we only want to use the errors `InvalidRequest/Response` for deserialization problems, we could also reflect that in the name `UnDeserializeableRequest` or so :)

Reply:
Also: Do we have no potential errors when serializing?